src/jwtcookie/Authentication/JwtCookie/JwtCookieHandler.cs

```diff
@@ -101,9 +111,22 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
                     {
                         var issuers = new[] { configuration.Issuer };
                         validationParameters.ValidIssuers = validationParameters.ValidIssuers?.Concat(issuers) ?? issuers;
-
-                        validationParameters.IssuerSigningKeys =
-                            validationParameters.IssuerSigningKeys?.Concat(configuration.SigningKeys) ?? configuration.SigningKeys;
+                        string issuer = _validator.ReadJwtToken(token).Issuer;
+
+                        if (issuer != null && _oidcProviderSettings.Count > 0)
+                        {
+                            foreach (KeyValuePair<string, OidcProvider> provider in _oidcProviderSettings)
+                            {
+                                if (provider.Value.Issuer == issuer)
+                                {
+                                    validationParameters.IssuerSigningKeys = await GetSigningKeys(provider.Value.WellKnownConfigEndpoint);
+                                }
+                            }
+                        }
+                        else
+                        {
+                            validationParameters.IssuerSigningKeys = validationParameters.IssuerSigningKeys?.Concat(configuration.SigningKeys) ?? configuration.SigningKeys;
+                        }
                     }
                 }
 
```

Comment:
Should we fallback to the default if there is no match in the loop for issuer

Reply:
@TheTechArch  Updated this piece of the code, is this what you mean?